src/test/java/commandparser/StringParserTest.java

```diff
@@ -0,0 +1,165 @@
+package commandparser;
+
+import data.exceptions.TaskManagerException;
+import org.junit.jupiter.api.Test;
+import time.MonthView;
+import time.WeekView;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+
+class StringParserTest {
+
+    @Test
+    void parseDateValidFormat() {
+        String validDateString = "15/03/2023";
+        LocalDate expectedDate = LocalDate.of(2023, 3, 15);
+        assertDoesNotThrow(() -> {
+            LocalDate parsedDate = StringParser.parseDate(validDateString);
+            assertEquals(expectedDate, parsedDate);
+        });
+    }
+
+    @Test
+    void parseDateInvalidFormat() {
+        String invalidDateString = "03-15-2023";
+        assertThrows(TaskManagerException.class, () -> StringParser.parseDate(invalidDateString));
```

Comment:
Great that both positive and negative test cases are considered for most features